### PR TITLE
fix(service) sce sanitation with fallback language

### LINKED ISSUE #1618: Sanitation strategy sce fails with fallback language
### Sanitation strategy `sce` fails with fallback language
When using a fallback language in conjuction with the sanitation strategy `sce` a type error is thrown when the string to translate is only available in the fallback language file.

Demo: https://plnkr.co/edit/iWdsq62cnGAY3mdN3aTM?p=preview

The `fallbackTranslationInstant` function expects a string as result but it gets a `TrustedValueHolderType` object.

```
typeError: result.substr is not a function
    at getFallbackTranslationInstant (angular-translate.js:1791)
    at resolveForFallbackLanguageInstant (angular-translate.js:1893)
    at fallbackTranslationInstant (angular-translate.js:1924)
    at determineTranslationInstant (angular-translate.js:2019)
    at Function.$translate.instant (angular-translate.js:2551)
    at translateFilter (angular-translate.js:3620)
    at fn (eval at compile (angular.js:14817), <anonymous>:4:97)
    at oneTimeInterceptedExpression (angular.js:16046)
    at oneTimeWatch (angular.js:15984)
    at Scope.$digest (angular.js:17515)
```

### Your environment
* version of angular-translate: 2.12.1
* version of angular: 1.5.8
* which browser and its version: Chrome 53.0.2785.143 m

### Steps to reproduce
- Two languages
- Set one langauge as fallback for another
- Set sanitation strategy to `sce`
- Translate a string via a filter

Demo: https://plnkr.co/edit/iWdsq62cnGAY3mdN3aTM?p=preview

### Expected behaviour
Everything should be nicely translated

### Actual behaviour
TypeError: `TrustedValueHolderType` object instead of expected `string`

### PR DESCRIPTION
**Pull request by MarijnMensinga**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **canary branch** (left side). Also you should start *your branch* off *our canary*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Description
`result` is assumed to be string but can also be object of type `TrustedValueHolderType`
closes #1618

### CODE REVIEW

**Comment by tspaeth** [comment]
What I'm missing: A testcase for this scenario

**Comment by MarijnMensinga** [comment]
@tspaeth unfortunately I'm not deep enough in the code of angular-translate to create a testcase/unittest. The only thing I know is that these steps need to be taken to reproduce the problem:

1. Use two different languages (eg. `EN` and `DE`)
2. Set one langauge as fallback for another (e.g. `EN` is fallback)
3. Select current language (e.g. `DE` is current language)
4. Set sanitation strategy to `sce` 
5. Translate a string that only exists in the fallback language via filter (e.g. `{{"hello"|translate}}`)
6. The error will now occur

Demo: https://plnkr.co/edit/iWdsq62cnGAY3mdN3aTM?p=preview

**Comment by knalli** [comment]
Well, this fixes the symptom but not the actual issue. In both cases (before the patch, after the patch) the actual feature in use (naming linking to another translation) will not work anymore.

**Comment by knalli** [comment]
And sorry: In `canary` landed a conflicting change.

File: src/service/translate.js [hunk 1]
```diff
@@ -1406,7 +1406,7 @@ function $translate($STORAGE_KEY, $windowProvider, $translateSanitizationProvide
           Interpolator.setLocale(langKey);
           result = Interpolator.interpolate(translationTable[translationId], interpolateParams, 'filter', sanitizeStrategy);
           result = applyPostProcessing(translationId, translationTable[translationId], result, interpolateParams, langKey, sanitizeStrategy);
-          if (result.substr(0, 2) === '@:') {
+          if (typeof result === 'string' && result.substr(0, 2) === '@:') {
```
Comment (knalli) [inline src/service/translate.js]: Perhaps calling resut.$$unwrapTrustedValue()?

**Comment by knalli** [comment]
Replaced with aaaa972

Thank you anyway!

**Comment by MarijnMensinga** [comment]
Thanks for the permanent fix. I already thought mine wouldn't be enough.

**Comment by tommck** [comment]
So.. you allow PRs with zero unit tests?

**Pull request closed without merging**